purchase_last_price_info/models/product_template.py

```diff
@@ -0,0 +1,54 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+# Copyright 2019 ForgeFlow S.L.
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    last_purchase_line_ids = fields.One2many(
+        comodel_name="purchase.order.line",
+        related="product_variant_ids.last_purchase_line_ids",
+        string="Last Purchase Order Lines",
+    )
+    last_purchase_line_id = fields.Many2one(
+        comodel_name="purchase.order.line",
+        compute="_compute_last_purchase_line_id",
+        string="Last Purchase Line",
+        store=True,
+    )
+    last_purchase_price = fields.Float(
+        related="last_purchase_line_id.price_unit", string="Last Purchase Price"
+    )
+    last_purchase_date = fields.Datetime(
+        related="last_purchase_line_id.date_order", string="Last Purchase Date"
+    )
+    last_purchase_supplier_id = fields.Many2one(
+        comodel_name="res.partner",
+        related="last_purchase_line_id.partner_id",
+        string="Last Supplier",
+    )
+    last_purchase_currency_id = fields.Many2one(
+        comodel_name="res.currency",
+        related="last_purchase_line_id.currency_id",
+        string="Last Purchase Currency",
+    )
+    show_last_purchase_price_currency = fields.Boolean(
+        related="product_variant_ids.show_last_purchase_price_currency",
+    )
+    last_purchase_price_currency = fields.Float(
+        string="Last currency purchase price",
+        related="product_variant_ids.last_purchase_price_currency",
+        digits=0,
+    )
+
+    @api.depends("last_purchase_line_ids")
+    def _compute_last_purchase_line_id(self):
+        for item in self:
+            if item.last_purchase_line_ids:
+                item.last_purchase_line_id = item.last_purchase_line_ids[:1]
+            else:
+                item.last_purchase_line_id = item.last_purchase_line_id
```

Comment:
No need to do all of this, and better to use this for not losing cache:
```
from odoo.fields import first
```

```suggestion
            item.last_purchase_line_id = first(item.last_purchase_line_ids)
```